src/shared/utilities/timeoutUtils.ts

```diff
@@ -39,44 +53,61 @@ export class Timeout {
         return remainingTime > 0 ? remainingTime : 0
     }
 
+    /**
+     * True when the Timeout has completed
+     */
+    public get fulfilled(): boolean {
```

Comment:
should this be renamed to `completed` ?

Reply:
I used fulfilled since I was referring to the underlying Promise. Maybe `terminated` to differentiate from `complete`?